docs/develop/python/message-passing.mdx

```diff
@@ -130,76 +142,95 @@ class GreetingWorkflow:
+
+- About validators:
+  - Use validators to reject an Update before it is written to History.
+    Validators are always optional.
+    If you don't need to reject updates, you can skip them.
+  - The SDK automatically provides a validator decorator named `@<update-handler-name>.validator`.
+    The validator must accept the same argument types as the handler and return `None`.
+    You can create your own validator with the same constraints by marking a method with the `@update_workflow_status.validator` decorator.
+  - For more flexibility, use code-based checks at the start of the Update handler instead of validators.
+    Raise an `ApplicationError` for any issues that would prevent the Update from running.
+    This approach adds rejected Update requests to your Event History if exceptions occur.
 
-- Validators are optional; if you don't want to be able to reject updates then you don't need a validator.
+- Accepting and rejecting Updates with validators:
+  - To reject an Update, raise an exception of any type in the validator.
+  - Without a validator, Updates are always accepted.
+- Validators and Event History:
+  - The `WorkflowExecutionUpdateAccepted` event is written into the History whether the acceptance was automatic or programmatic.
+  - When a Validator raises an error, the Update is rejected and `WorkflowExecutionUpdateAccepted` _won't_ be added to the Event History.
+    The caller receives an "Update failed" error.
 
-- The name of the decorator you use to define the validator is based on the name that you give to the handler.
-  The validator must take the same argument type as the handler, but always returns `None`.
+- Update (and Signal) handlers can be `async def`, letting them use Activities, Child Workflows, durable [`asyncio.sleep`](https://docs.python.org/3/library/asyncio-task.html#asyncio.sleep) Timers, [`workflow.wait_condition`](https://python.temporal.io/temporalio.workflow.html#wait_condition) conditions, and more.
+  See [Async handlers](#async-handlers) and [Workflow message passing](/encyclopedia/workflow-message-passing) for safe usage guidelines.
 
-- To reject an Update, you raise an exception (of any type) in the validator.
+## Send messages {#send-messages}
 
-- Update (and Signal) handlers can also be `async def` and thus use Activities, Child Workflows, durable [`asyncio.sleep`](https://docs.python.org/3/library/asyncio-task.html#asyncio.sleep) timers, [`workflow.wait_condition`](https://python.temporal.io/temporalio.workflow.html#wait_condition) conditions, etc. See [Async handlers](#async-handlers) below, and read [Workflow message passing](/encyclopedia/workflow-message-passing) for an introduction to safe usage of async Signal and Update handlers.
+To send Queries, Signals, or Updates, you call methods on a [WorkflowHandle](https://python.temporal.io/temporalio.client.WorkflowHandle.html) object.
 
+To fetch the Workflow handle:
```

Comment:
Not "fetch" here: `get_workflow_handle_for` is a plain `def` and therefore does not do any network call.

Reply:
Removed this line, added colon to previous line